odoo/addons/base/models/ir_cron.py

```diff
@@ -530,3 +589,18 @@ class ir_cron_trigger(models.Model):
     @api.autovacuum
     def _gc_cron_triggers(self):
         self.search([('call_at', '<', datetime.now() + relativedelta(weeks=-1))]).unlink()
+
+
+class ir_cron_progress(models.Model):
+    _name = 'ir.cron.progress'
+    _description = 'Actions Progress'
+    _rec_name = 'cron_id'
+
+    cron_id = fields.Many2one("ir.cron", index=True)
+    remaining = fields.Integer(default=0)
+    done = fields.Integer(default=0)
+    timed_out = fields.Boolean(default=False)
```

Comment:
It could be `default=True`

Reply:
Chose not to go that route and instead be explicit in the `_process_job`.
If this method is ever used elsewhere it should not use the timed_out tag and we want to avoid having this king of mistake

Reply:
I personally prefer defaults in this case.